scvi/models/modules.py

```diff
@@ -273,3 +330,129 @@ def forward(self, x: torch.Tensor, *cat_list: int):
         p_m = self.mean_decoder(p)
         p_v = torch.exp(self.var_decoder(p))
         return p_m, p_v
+
+
+class MultiEncoder(nn.Module):
+    def __init__(
+        self,
+        n_heads: int,
+        n_input_list: List[int],
+        n_output: int,
+        n_hidden: int = 128,
+        n_layers_individual: int = 1,
+        n_layers_shared: int = 2,
+        n_cat_list: Iterable[int] = None,
+        dropout_rate: float = 0.1,
+    ):
+        super().__init__()
+
+        self.encoders = ModuleList(
+            [
+                FCLayers(
+                    n_in=n_input_list[i],
+                    n_out=n_hidden,
+                    n_cat_list=n_cat_list,
+                    n_layers=n_layers_individual,
+                    n_hidden=n_hidden,
+                    dropout_rate=dropout_rate,
+                    n_batch_norm=1,
+                )
+                for i in range(n_heads)
+            ]
+        )
+
+        self.encoder_shared = FCLayers(
+            n_in=n_hidden,
+            n_out=n_hidden,
+            n_cat_list=n_cat_list,
+            n_layers=n_layers_shared,
+            n_hidden=n_hidden,
+            dropout_rate=dropout_rate,
+        )
+
+        self.mean_encoder = nn.Linear(n_hidden, n_output)
+        self.var_encoder = nn.Linear(n_hidden, n_output)
+
+    def forward(self, x: torch.Tensor, head_id: int, *cat_list: int):
+        q = self.encoders[head_id](x, *cat_list)
+        q = self.encoder_shared(q, *cat_list)
+
+        q_m = self.mean_encoder(q)
+        q_v = torch.exp(self.var_encoder(q))
+        latent = reparameterize_gaussian(q_m, q_v)
```

Comment:
do we really need this reparameterize gaussian ? not a critical point

Reply:
Not sure I get your question, yes it is the reparametrization trick for variational inference.

Reply:
I mean, do we need the function called reparameterize_gaussian ? Instead of torch.Normal(...).rsample() ?

Reply:
Imo, it increases the code readability: the reader understand that this line is executed to "reparameterize the gaussian".

Reply:
good to me